security/jwt/src/test/java/io/quarkus/ts/security/jwt/BaseJwtSecurityIT.java

```diff
@@ -13,9 +13,8 @@
 import org.junit.jupiter.api.Test;
 
 import io.restassured.http.ContentType;
-import io.restassured.specification.RequestSpecification;
 
-public abstract class BaseJwtSecurityIT {
+public abstract class BaseJwtSecurityIT extends BaseJwtSecurityGraphQLIT {
```

Comment:
I understand what we want to do here (thanks to the excellent PR description :+1:), but I have 2 objections:
- The hierarchy of the test classes is a bit confusing. There is a GraphQL-related class as a parent to a more generic test class, weird placement of an abstract method, a parent calling a static method of a child.
- We perform only one dev mode test, tailor made to the GraphQL scenario. Why not run all the existing tests in dev mode?

My suggestion is:
- Get rid of `BaseJwtSecurityGraphQLIT`.
  - Move `protected abstract RequestSpecification givenWithToken` back to `BaseJwtSecurityIT` together with both new methods. I.e. make `testRouteContextTermination` part of the `BaseJwtSecurityIT` suite.
- Make `DevModeJwtSecurityGraphQLIT` extend `BaseJwtSecurityIT`.
  - And also maybe replace `RestAssured.given().auth().oauth2(token)` call with `app.given().auth().oauth2(token)` to make it more obvious. :slightly_smiling_face:

That way, the test structure will be more readable and we increase the coverage (+29 tests).
`DevModeJwtSecurityGraphQLIT` is not `fips-incompatible`, so we don't have to worry about that (I tested it on FIPS machine).

What do you think?

Reply:
I agree with suggestions, thank you. Done.